src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final String REPORT_SEPARATOR = " - ";
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate;
+        LocalDate endDate;
+        try {
+            startDate = LocalDate.parse(dateFrom, formatter);
+            endDate = LocalDate.parse(dateTo, formatter);
+        } catch (DateTimeParseException exception) {
+            throw new IllegalArgumentException("Wrong date formats.");
+        }
+        String reportHeader =
+                String.format("Report for period %1$s - %2$s", startDate.format(formatter),
+                        endDate.format(formatter));
+        StringBuilder report = new StringBuilder(reportHeader);
+        for (String name : names) {
+            int employeeSalary = 0;
+            for (String dataRecord : data) {
+                if (!dataRecord.contains(name)) {
+                    continue;
+                }
+                final int dataEmployeeWorkingHoursIndex = 2;
+                final int dataEmployeeIncomePerHourIndex = 3;
+                LocalDate parsedDate;
+                String[] dataRecordSplit = dataRecord.split(" ");
+                String employeeWorkingHours = dataRecordSplit[dataEmployeeWorkingHoursIndex];
+                String employeeIncomePerHour = dataRecordSplit[dataEmployeeIncomePerHourIndex];
+                try {
+                    parsedDate = LocalDate.parse(dataRecordSplit[0], formatter);
+                    if (parsedDate.toEpochDay() < startDate.toEpochDay()
+                            || parsedDate.toEpochDay() > endDate.toEpochDay()) {
+                        continue;
```

Comment:
use isBefore/isAfter methods to make the code easier to understand.
And invert IF condition to avoid using **continue**;